### DISCUSSION TITLE
运行matcher时发生神秘报错 我看不懂 所以是神秘报错（

### DISCUSSION THREAD

**Discussion by shoucandanghehe** [nonebot/discussions#477] [问答] [answered]
```
07-13 00:57:38 [SUCCESS] nonebot | OneBot V11 *** | [message.group.normal]: Message -*** from ***@[群:***] '同意重启'
07-13 00:57:38 [DEBUG] nonebot | Running PreProcessors...
07-13 00:57:38 [DEBUG] nonebot | Checking for matchers in priority 0...
07-13 00:57:38 [DEBUG] nonebot | Checking for matchers in priority 1...
07-13 00:57:38 [DEBUG] nonebot | Plugin-Alconna | Alconna::reboot 对 "同意重启" 的解析结果是 (matched=True, header_match=(origin='reboot' result='reboot' matched=True groups={}), options={'agree': (value=Ellipsis args={})})
07-13 00:57:38 [INFO] nonebot | Event will be handled by AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75)
07-13 00:57:38 [DEBUG] nonebot | Running AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75)
07-13 00:57:38 [DEBUG] nonebot | Running handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),))
07-13 00:57:38 [DEBUG] nonebot | Handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),)) skipped
07-13 00:57:38 [DEBUG] nonebot | Running handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),))
07-13 00:57:38 [DEBUG] nonebot | OneBot V11 | Calling API send_msg
07-13 00:57:38 [DEBUG] nonebot | Running CalledAPI hooks...
07-13 00:58:05 [DEBUG] nonebot | Running PreProcessors...
07-13 00:58:05 [INFO] nonebot | AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75) running complete
07-13 00:58:05 [ERROR] nonebot | Running AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75) failed.
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 277, in _init_core_attrs
    self._core_schema = _getattr_no_parents(self._type, '__pydantic_core_schema__')
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 119, in _getattr_no_parents
    raise AttributeError(attribute)
AttributeError: __pydantic_core_schema__

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen os>", line 711, in __getitem__
KeyError: 'PYDANTIC_SKIP_VALIDATING_CORE_SCHEMAS'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 796, in ensure_context
    yield
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 825, in simple_run
    await handler(
GeneratorExit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 15, in <module>
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\__init__.py", line 335, in run
    get_driver().run(*args, **kwargs)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
    uvicorn.run(
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\uvicorn\main.py", line 577, in run
    server.run()
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\uvicorn\server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\ProgramData\miniconda3\Lib\asyncio\runners.py", line 194, in run
    return runner.run(main)
  File "C:\ProgramData\miniconda3\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "C:\ProgramData\miniconda3\Lib\asyncio\base_events.py", line 674, in run_until_complete
    self.run_forever()
  File "C:\ProgramData\miniconda3\Lib\asyncio\windows_events.py", line 322, in run_forever
    super().run_forever()
  File "C:\ProgramData\miniconda3\Lib\asyncio\base_events.py", line 641, in run_forever
    self._run_once()
  File "C:\ProgramData\miniconda3\Lib\asyncio\base_events.py", line 1987, in _run_once
    handle._run()
  File "C:\ProgramData\miniconda3\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\dependencies\__init__.py", line 177, in _solve_field
    v = check_field_type(field, value)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\dependencies\utils.py", line 54, in check_field_type
    return model_field_validate(field, value, DEFAULT_CONFIG)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\compat.py", line 172, in model_field_validate
    return TypeAdapter(
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 264, in __init__
    self._init_core_attrs(rebuild_mocks=False)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 142, in wrapped
    return func(self, *args, **kwargs)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 284, in _init_core_attrs
    self._core_schema = _get_schema(self._type, config_wrapper, parent_depth=self._parent_depth)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 103, in _get_schema
    schema = gen.clean_schema(schema)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 452, in clean_schema
    schema = validate_core_schema(schema)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\_internal\_core_utils.py", line 566, in validate_core_schema
    if 'PYDANTIC_SKIP_VALIDATING_CORE_SCHEMAS' in os.environ:
  File "<frozen _collections_abc>", line 813, in __contains__
  File "<frozen os>", line 714, in __getitem__
> File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\message.py", line 428, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 850, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 815, in simple_run
    with self.ensure_context(bot, event):
  File "C:\ProgramData\miniconda3\Lib\contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 798, in ensure_context
    current_bot.reset(b_t)
ValueError: <Token var=<ContextVar name='current_bot' at 0x000002AC048A3BA0> at 0x000002AC081DE5C0> was created in a different Context
Exception ignored in: <coroutine object check_and_run_matcher at 0x000002AC07E1E790>
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\message.py", line 476, in check_and_run_matcher
    await _run_matcher(
RuntimeError: coroutine ignored GeneratorExit
Task was destroyed but it is pending!
task: <Task pending name='Task-41282' coro=<Bot.handle_event() running at C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\adapters\onebot\v11\bot.py:210> wait_for=<_GatheringFuture pending cb=[Task.task_wakeup()]>>
```

**Comment by yanyongyu** [comment]
@RF-Tar-Railt

**Comment by RF-Tar-Railt** [comment]
> ```
> 07-13 00:57:38 [SUCCESS] nonebot | OneBot V11 *** | [message.group.normal]: Message -*** from ***@[群:***] '同意重启'
> 07-13 00:57:38 [DEBUG] nonebot | Running PreProcessors...
> 07-13 00:57:38 [DEBUG] nonebot | Checking for matchers in priority 0...
> 07-13 00:57:38 [DEBUG] nonebot | Checking for matchers in priority 1...
> 07-13 00:57:38 [DEBUG] nonebot | Plugin-Alconna | Alconna::reboot 对 "同意重启" 的解析结果是 (matched=True, header_match=(origin='reboot' result='reboot' matched=True groups={}), options={'agree': (value=Ellipsis args={})})
> 07-13 00:57:38 [INFO] nonebot | Event will be handled by AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75)
> 07-13 00:57:38 [DEBUG] nonebot | Running AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75)
> 07-13 00:57:38 [DEBUG] nonebot | Running handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),))
> 07-13 00:57:38 [DEBUG] nonebot | Handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),)) skipped
> 07-13 00:57:38 [DEBUG] nonebot | Running handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),))
> 07-13 00:57:38 [DEBUG] nonebot | OneBot V11 | Calling API send_msg
> 07-13 00:57:38 [DEBUG] nonebot | Running CalledAPI hooks...
> 07-13 00:58:05 [DEBUG] nonebot | Running PreProcessors...
> 07-13 00:58:05 [INFO] nonebot | AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75) running complete
> 07-13 00:58:05 [ERROR] nonebot | Running AlconnaMatcher(type='', module=nonebot_plugin_zomboid, lineno=75) failed.
> Traceback (most recent call last):
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 277, in _init_core_attrs
>     self._core_schema = _getattr_no_parents(self._type, '__pydantic_core_schema__')
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 119, in _getattr_no_parents
>     raise AttributeError(attribute)
> AttributeError: __pydantic_core_schema__
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "<frozen os>", line 711, in __getitem__
> KeyError: 'PYDANTIC_SKIP_VALIDATING_CORE_SCHEMAS'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 796, in ensure_context
>     yield
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 825, in simple_run
>     await handler(
> GeneratorExit
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "<string>", line 15, in <module>
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\__init__.py", line 335, in run
>     get_driver().run(*args, **kwargs)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
>     uvicorn.run(
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\uvicorn\main.py", line 577, in run
>     server.run()
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\uvicorn\server.py", line 65, in run
>     return asyncio.run(self.serve(sockets=sockets))
>   File "C:\ProgramData\miniconda3\Lib\asyncio\runners.py", line 194, in run
>     return runner.run(main)
>   File "C:\ProgramData\miniconda3\Lib\asyncio\runners.py", line 118, in run
>     return self._loop.run_until_complete(task)
>   File "C:\ProgramData\miniconda3\Lib\asyncio\base_events.py", line 674, in run_until_complete
>     self.run_forever()
>   File "C:\ProgramData\miniconda3\Lib\asyncio\windows_events.py", line 322, in run_forever
>     super().run_forever()
>   File "C:\ProgramData\miniconda3\Lib\asyncio\base_events.py", line 641, in run_forever
>     self._run_once()
>   File "C:\ProgramData\miniconda3\Lib\asyncio\base_events.py", line 1987, in _run_once
>     handle._run()
>   File "C:\ProgramData\miniconda3\Lib\asyncio\events.py", line 88, in _run
>     self._context.run(self._callback, *self._args)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\dependencies\__init__.py", line 177, in _solve_field
>     v = check_field_type(field, value)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\dependencies\utils.py", line 54, in check_field_type
>     return model_field_validate(field, value, DEFAULT_CONFIG)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\compat.py", line 172, in model_field_validate
>     return TypeAdapter(
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 264, in __init__
>     self._init_core_attrs(rebuild_mocks=False)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 142, in wrapped
>     return func(self, *args, **kwargs)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 284, in _init_core_attrs
>     self._core_schema = _get_schema(self._type, config_wrapper, parent_depth=self._parent_depth)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\type_adapter.py", line 103, in _get_schema
>     schema = gen.clean_schema(schema)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 452, in clean_schema
>     schema = validate_core_schema(schema)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\pydantic\_internal\_core_utils.py", line 566, in validate_core_schema
>     if 'PYDANTIC_SKIP_VALIDATING_CORE_SCHEMAS' in os.environ:
>   File "<frozen _collections_abc>", line 813, in __contains__
>   File "<frozen os>", line 714, in __getitem__
> > File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\message.py", line 428, in _run_matcher
>     await matcher.run(bot, event, state, stack, dependency_cache)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 850, in run
>     await self.simple_run(bot, event, state, stack, dependency_cache)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 815, in simple_run
>     with self.ensure_context(bot, event):
>   File "C:\ProgramData\miniconda3\Lib\contextlib.py", line 158, in __exit__
>     self.gen.throw(value)
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 798, in ensure_context
>     current_bot.reset(b_t)
> ValueError: <Token var=<ContextVar name='current_bot' at 0x000002AC048A3BA0> at 0x000002AC081DE5C0> was created in a different Context
> Exception ignored in: <coroutine object check_and_run_matcher at 0x000002AC07E1E790>
> Traceback (most recent call last):
>   File "C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\message.py", line 476, in check_and_run_matcher
>     await _run_matcher(
> RuntimeError: coroutine ignored GeneratorExit
> Task was destroyed but it is pending!
> task: <Task pending name='Task-41282' coro=<Bot.handle_event() running at C:\Users\Administrator\Desktop\bot\.venv\Lib\site-packages\nonebot\adapters\onebot\v11\bot.py:210> wait_for=<_GatheringFuture pending cb=[Task.task_wakeup()]>>
> ```

？？

**Comment by RF-Tar-Railt** [comment]
python版本，nb版本，插件版本？

**Comment by shoucandanghehe** [comment, marked as answer]
找到原因了 是因为rcon发命令带空格没加双引号导致调用没有正常返回 但是不知道为什么报错全是nb的）
